### PR TITLE
[cudnn] Update to cudnn 8.x 

### LINKED ISSUE #16636: cudnn support for 8.x
With the new RTX 30x0 GPU family it is necessary to use **CUDA 11.x + cudnn 8.x**
Currently, building opencv4 with the cuda backend fails due to the cudnn port being referenced to cudnn 7.6

I can build opencv4 modyfing (quickly and dirty) the cudnn port with these major changes:

- downloading the package from [https://anaconda.org/conda-forge/cudnn/8.1.0.77/download/win-64/cudnn-8.1.0.77-h3e0f4f4_0.tar.bz2](https://anaconda.org/conda-forge/cudnn/8.1.0.77/download/win-64/cudnn-8.1.0.77-h3e0f4f4_0.tar.bz2)
`    set(CUDNN_DOWNLOAD_LINK "https://anaconda.org/conda-forge/cudnn/8.1.0.77/download/win-64/cudnn-8.1.0.77-h3e0f4f4_0.tar.bz2")
    set(SHA512_CUDNN "820fcd1986eba673f2821f16c20c4ae97f444488d3e03ba11b307e59678b0e141ed39235f0396433b3c63e2d901902a8096175fe91391be24104e9eb5dcb9cb1")
    set(CUDNN_OS "windows")`
- With cuda 11 Nvidia splits header and lib files across multiple files. It's now required to install all of them 
`    file(INSTALL "${SOURCE_PATH}/Library/include/" DESTINATION ${CURRENT_PACKAGES_DIR}/include FILES_MATCHING PATTERN "*.h")
    file(INSTALL "${SOURCE_PATH}/Library/lib/" DESTINATION ${CURRENT_PACKAGES_DIR}/lib FILES_MATCHING PATTERN "*.lib")
    file(INSTALL "${SOURCE_PATH}/Library/bin/" DESTINATION ${CURRENT_PACKAGES_DIR}/bin FILES_MATCHING PATTERN "*.dll")`

- Nvidia rename license and change path.
`file(INSTALL "${SOURCE_PATH}/info/licenses/NVIDIA_SLA_cuDNN_Support.txt" DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`

I hope this could help in speeding up the update of portfile in order to make it working with new cuda & cudnn

### PR DESCRIPTION
**Pull request by IlTeo285**
Follow-up of:
  Fixes #16636

**Description**

- Insert the check for `CUDA_PATH` environmental variable the is currently set by CUDA installer
- If cudnn is not present, the download is performed using conda-forge cudnn 8

### CODE REVIEW

**Comment by IlTeo285** [comment]
All the tests failed due to pipelines running on CUDA 10.1

File: ports/cudnn/CONTROL [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Source: cudnn
-Version: 7.6
-Port-Version: 4
+Version: 8.1
+Port-Version: 5
```
Comment (NancyLi1013) [inline ports/cudnn/CONTROL:3]:
```suggestion
```
For a new version update, this filed is not needed here.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/cudnn/portfile.cmake [hunk 2]
```diff
@@ -84,18 +86,17 @@ else()
       NO_REMOVE_ONE_LEVEL
   )
 
+
```
Comment (NancyLi1013) [inline ports/cudnn/portfile.cmake:89]:
```suggestion
```

**Review by NancyLi1013 — COMMENTED** [review]
Could you please also update CONTROL file for `cuda`?

**Comment by NancyLi1013** [comment]
@cenit 

Could you please help review this PR? Thanks.

**Comment by cenit** [comment]
is it possible to build this on top of #16413 which is still unmerged?

**Comment by NancyLi1013** [comment]
Thanks for your suggestion @cenit.

**Comment by IlTeo285** [comment]
@NancyLi1013I think that if #16413 mergeing is imminent enough,  I could rebase this to master as soon as the merge is complete.

**Comment by IlTeo285** [comment]
@NancyLi1013 Sorry for bothering, is it the file modification review still pendand because i miss some steps?
I see the system claiming "1 change request" but it seems to me already changed.

**Comment by NancyLi1013** [comment]
@IlTeo285 

Currently, there is nothing needs to be done from your side. You can rebase this PR after PR #16413 merged.

**Comment by NancyLi1013** [comment]
@IlTeo285 

Could you please rebase this PR since PR #16413 has been merged?

**Comment by NancyLi1013** [comment]
@IlTeo285 

Could you please revert the unrelated changes in this PR?

**Comment by NancyLi1013** [comment]
Please open a new PR to submit your changes if you are still interested in this port. Since there are many unrelated changes merged to this PR.

**Pull request closed without merging**